qp/utils.py

```diff
@@ -362,9 +362,10 @@ def interpolate_multi_x_multi_y(x, xvals, yvals, **kwargs):
     xy_vals = np.hstack([xvals, yvals])
     nx = xvals.shape[-1]
     def single_row(xy_vals_):
+        print(xy_vals_[0:nx].shape,xy_vals_[nx:].shape)
         return interp1d(xy_vals_[0:nx], xy_vals_[nx:], **kwargs)(x)
     vv = np.vectorize(single_row, signature="(%i)->(%i)" % (xvals.shape[-1]+yvals.shape[-1], x.size))
-    return vv(xy_vals)
+    return  vv(xy_vals).flatten()
```

Comment:
@eacharles I had to add the flattening here to pass the unit test. And the flattening for the factored case is present in the hist version of this code, previously used in quant_pdf. Still, can you cross-check me here, and if correct check the docstrings of the rest of the code, as I think these are copy-paste all the way down

Reply:
I should probably sit down at some point and figure out exactly what is going on with the array shaping and flattening and write it all down.  I agree that it is confusing.   But this looks good for now.